core/src/main/java/org/apache/accumulo/fate/zookeeper/ZooCache.java

```diff
@@ -435,6 +459,44 @@ public T retry() {
     return zr.retry();
   }
 
+  /*
+   * Some of the table configuration parameters are possibly not in ZooKeeper yet and even though
+   * you are able you should not put a watcher on them to observe the NodeCreated event. If you do
+   * that and the ZNode for that configuration is never created, you will have no way to delete the
+   * watcher you created for it in Zookeeper. Unremovable watchers that are created for for table
+   * configs that never actually have a Znode is causing memory leaks in Accumulo instances that
+   * create and delete tables at a high rate.
+   */
+
+  private boolean isWatched(ZooKeeper zooKeeper, String zPath)
+      throws KeeperException, InterruptedException {
+
+    boolean watched = true;
+
+    if (znodeExists.containsKey(zPath))
+      return watched;
+
+    if (zPath.contains("tserver.") || zPath.contains("table.")) {
```

Comment:
This check seems odd to me.... are we looking for only configuration properties here?

Reply:
I know that we agreed to change the zPath.contains("tserver") ... to a regex, but you hit on something about just trying to get it to work for all zPaths which is a just cause.  I gave it a shot and it broke a lot of the ZooCacheTests but it is because different function calls need to be expected in the Mock framework.  The Uno instance looked like it was doing just fine.  I can keep working this idea if you want me too or just go with what we agreed on.